tests/func/test_checkout.py

```diff
@@ -581,3 +584,52 @@ def test_checkout_no_checksum(repo_dir, dvc_repo):
     stage = dvc_repo.run(outs=[repo_dir.FOO], no_exec=True, cmd="somecmd")
     dvc_repo.checkout(stage.path, force=True)
     assert not os.path.exists(repo_dir.FOO)
+
+
+def test_checkout_should_avoid_copy_when_adding_for_existing_cache(
+    repo_dir, dvc_repo, caplog
+):
+    dvc_repo.cache.local.cache_types = ["copy"]
+
+    dvc_repo.add(repo_dir.FOO)
+    dvc_repo.add(repo_dir.BAR)
+
+    shutil.copy(repo_dir.BAR, repo_dir.FOO)
+    foo_inode = get_inode(repo_dir.FOO)
+
+    caplog.clear()
+    with caplog.at_level(logging.INFO):
+        dvc_repo.add(repo_dir.FOO)
+
+    assert "Removing before checkout" not in caplog.text
+    assert get_inode(repo_dir.FOO) == foo_inode
```

Comment:
Inodes are reused so this is not a reliable test.

Reply:
check removed